ethcore/machine/src/externalities.rs

```diff
@@ -479,6 +479,7 @@ mod tests {
 			last_hashes: Arc::new(vec![]),
 			gas_used: 0.into(),
 			gas_limit: 0.into(),
+			chain_id: 0.into(),
```

Comment:
We can avoid changing the structure altogether. `Externalities` contains a reference to `Machine`, which allows retrieving the complete config parameter via `Machine::params` already. What we may be able to do here to simplify thing is to create a new function in externalities, and then make chain id opcode call that externalities.

Reply:
You mean extending [`vm::Ext` trait](https://github.com/paritytech/parity-ethereum/blob/141f6a047ece33cf54bea130707553a075d152b2/ethcore/vm/src/ext.rs#L67) to include a `chain_id()` method and then implement it on [`machine::Externalities`](https://github.com/paritytech/parity-ethereum/blob/bcd43d6b67b62507ca8429f0e1f5a63ae735eeef/ethcore/machine/src/externalities.rs#L131) using the ref to `Machine`?

Reply:
Yes exactly.